common/protob/messages-monero.proto

```diff
@@ -142,6 +142,7 @@ message MoneroTransactionInitRequest {
         optional uint32 client_version = 13;  // connected client version
         optional uint32 hard_fork = 14;       // transaction hard fork number
         optional bytes monero_version = 15;   // monero software version
+        optional uint32 sig_version = 16;     // MLSAG / CLSAG
```

Comment:
Is it really necessary to add a new field? Will the new fork support both MLSAG and CLSAG? If not, I think the `sig_version` value could be inferred unambiguously from the `hard_fork` field.

Reply:
it is probably not required. Depends on how the hard fork rules will be set when it comes to it. If certain hard fork only permits CLSAG but does not enforces them, then we want to leave the decision on the signing algorithm to the wallets. (Similar situation was with bulletproofs if I remember correctly).

If CLSAG usage will be determined purely by hard fork, the field is not needed.